beamer/health/check.py

```diff
@@ -0,0 +1,526 @@
+import logging
+from collections import defaultdict
+from dataclasses import dataclass, field
+from datetime import datetime
+from pathlib import Path
+from typing import DefaultDict, Optional, TypedDict, cast
+
+import toml
+from eth_typing import BlockNumber
+from typing_extensions import NotRequired
+from web3 import Web3
+from web3.constants import ADDRESS_ZERO
+from web3.middleware import geth_poa_middleware
+from web3.types import BlockData
+
+import beamer.agent.contracts
+import beamer.agent.events
+from beamer.agent.config import _merge_dicts
+from beamer.agent.events import ClaimMade, DepositWithdrawn, RequestCreated, RequestFilled
+from beamer.agent.typing import ChainId
+from beamer.agent.util import load_ERC20_abi, setup_logging
+from beamer.health.notification_state import NotificationState
+from beamer.health.notify import Message, NotificationConfig, Notify
+
+logging.disable()
+
+
+class MessageTypes:
+    REQUEST_EXPIRED = "RequestExpired"
+    DEPOSIT_WITHDRAWN = "DepositWithdrawn"
+    UNCLAIMED_FILL = "UnclaimedFill"
+    CHALLENGE_GAME = "ChallengeGame"
+    CHALLENGE_GAME_CLAIM_BY_SOMEONE_ELSE = "ChallengeGameClaimedBySomeoneElse"
+
+
+TokenMap = dict[str, list[tuple[str, str]]]
+
+
+class MonitorConfig(TypedDict):
+    agent_address: str
+    deployment_dir: Path
+    cache_file_path: Path
+    notification_system: str
+    rpcs: dict[int, tuple[str, BlockNumber]]
+    explorers: dict[int, str]
+    notification: NotificationConfig
+    tokens: TokenMap
+
+
+class Transfer(TypedDict):
+    created: NotRequired[RequestCreated]
+    filled: NotRequired[RequestFilled]
+    claimed: NotRequired[dict[int, list[ClaimMade]]]
+    withdrawn: NotRequired[DepositWithdrawn]
+
+
+TransferMap = DefaultDict[str, Transfer]
+
+ChainEventMap = dict[ChainId, list[beamer.agent.events.Event]]
+
+
+class NotificationMeta(TypedDict):
+    request_id: str
+    message_type: str
+    message_link: str
+
+
+class Notification(TypedDict):
+    meta: NotificationMeta
+    body: str
+
+
+@dataclass
+class Stats:
+    fills: int = 0
+    requests: int = 0
+    withdraws: int = 0
+    volume: int = 0
+
+
+@dataclass
+class TransferStats:
+    volume: int = 0
+    fills: int = 0
+    requests: int = 0
+    withdrawals: int = 0
+    claims: int = 0
+    expired_requests: int = 0
+    expired_requests_volume: int = 0
+
+
+@dataclass
+class Context:
+    stats: TransferStats = field(default_factory=TransferStats)
+    messages: list = field(default_factory=list)
+    agent_address: str = ""
+
+    notification_state: NotificationState = NotificationState(Path("chain/"))
+
+    def add_message(self, message: Notification) -> None:
+        self.messages.append(message)
+
+
+claim_request_extension = 86400
+global_config: Optional[MonitorConfig] = None
```

Comment:
Please use all-uppercase for such vars:
```python
GLOBAL_CONFIG: None | MonitorConfig = None
```

Note that I also used the new, preferred union typing syntax (`A | B`).